pykafka/balancedconsumer.py

```diff
@@ -460,22 +473,16 @@ def _path_from_partition(self, p):
     def _remove_partitions(self, partitions):
         """Remove partitions from the zookeeper registry for this consumer.
 
-        Also remove these partitions from the consumer's internal
-        partition registry.
-
         :param partitions: The partitions to remove.
         :type partitions: Iterable of :class:`pykafka.partition.Partition`
         """
         for p in partitions:
-            assert p in self._partitions
+            # TODO pass zk node version to make sure we still own this node
```

Comment:
Couldn't make up my mind about whether that's really worth it - we would have to emit node versions from `_get_held_partitions()` and pass those into this. Maybe a lower-priority thing for a separate ticket?